### PR TITLE
Fix seo_canonical option to be translated for the frontend, and some optimizations

### PR DESCRIPTION
**Pull request by texpert**
Currently, the canonical link, when it has different links on different languages, is generating a broken link:

```html
<link rel="canonical" href="[&lt;!--:en--&gt;https://florsan.md/&lt;!--:--&gt;&lt;!--:ru--&gt;https://florsan.md/ru&lt;!--:--&gt;&lt;!--:md--&gt;https://florsan.md/md&lt;!--:--&gt;](view-source:http://0.0.0.0:3000/%3C!--:en--%3Ehttps://florsan.md/%3C!--:--%3E%3C!--:ru--%3Ehttps://florsan.md/ru%3C!--:--%3E%3C!--:md--%3Ehttps://florsan.md/md%3C!--:--%3E)">
```

This PR is fixing it, calling `translate`: `current_site.get_option('seo_canonical')&.translate`

Also, some small performance optimizations were implemented.

### CODE REVIEW

File: app/helpers/camaleon_cms/frontend/application_helper.rb
Comment (brian-kephart) [inline app/helpers/camaleon_cms/frontend/application_helper.rb:27]: This refactoring is not equivalent. If `_current_site` is nil, then the `size` method will be called on `nil` and return a `NoMethodError`.

**Review by brian-kephart — CHANGES_REQUESTED** [review]
Thanks @texpert! These are excellent changes, especially the new test.

I found one potential bug and had a question about another issue. If those are addressed then it LGTM.

If the change in behavior is intentional, then we might want to merge this in the v3 branch, though since it's also a bugfix, maybe it should go to master. What do you think?

File: app/helpers/camaleon_cms/frontend/application_helper.rb
Comment (brian-kephart) [reply app/helpers/camaleon_cms/frontend/application_helper.rb:27]: Only `get_languages` won't be called. The safe navigation operator doesn't end the chain, it continues the chain with nil as the receiver of the next method.

```ruby
nil&.get_languages.size

# `<main>': undefined method `size' for nil:NilClass (NoMethodError)
```

It's equivalent to this:

```ruby
(_current_site.nil? ? nil : _current_site.get_languages).size
```

That's why RuboCop complains if you use safe navigation then continue the chain with regular dots instead of more safe navigation. The safe thing would be to put it back how it was, or do this:
```ruby
_current_site&.get_languages&.size&.> 1
```

Comment (texpert) [reply app/helpers/camaleon_cms/frontend/application_helper.rb]: Finally I have tried this in console and now I am understanding it!

Thanks, @brian-kephart !

**Comment by texpert** [comment]
@brian-kephart, issues from your code review have been addressed, thanks again!

**Comment by texpert** [comment]
I think this should go to master and then we can publish a new 2.x release.

**Comment by brian-kephart** [comment]
@texpert I agree. I have a couple of pending security fixes to go in the next release as well, so could you please leave it to me? I plan to release 2.6.5 on Monday 3/20 after hearing back regarding one of the fixes.

**Review by brian-kephart — APPROVED** [review]

**Comment by texpert** [comment]
> @texpert I agree. I have a couple of pending security fixes to go in the next release as well, so could you please leave it to me? I plan to release 2.6.5 on Monday 3/20 after hearing back regarding one of the fixes.

@brian-kephart, do you mean that I shouldn't merge this PR until your fixes will get in, or crafting a new release? I the latter, than I have no possibility of releasing the gem, so I won't go anyway for only a GitHub release.

**Comment by brian-kephart** [comment]
@texpert you can merge. I was referring to the release.

**Comment by texpert** [comment]
Also, @brian-kephart, how about the next release to be 2.7.0, not 2.6.5? Because we got rid of CoffeeScript, I think it is worth to be a minor, not a tiny release.

**Pull request merged**
